### PR TITLE
added a mapping check to consortia

### LINKED ISSUE #730: UI: add a mapping check
Screen Shot 2019-09-11 at 12 03 37 PM
# Task

It's not easy to tell when users are ready/mapped, putting a checkmark on the avatar picture seems like a good solution, see attached photo.

* To do this you'll need to add a column in the `consortia` db table, something like `mappedForRun`
* the users will update that table and add if they're mapped from the `maps` page
* unmap actions should also remove the user from that list, check out how unmap works on active pipeline change

### PR DESCRIPTION
**Pull request by kalitopix**
# Problem
referenced issue(s):

# Solution

# Testing

closes #730

### CODE REVIEW

**Comment by rssk** [comment]
Looks good! just add the unmap portion too

File: packages/coinstac-api-server/src/data/shared-fields.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ const schemaFields = {
     members: [ID]
     pipelines: [ID]
     results: [ID]
+    mappedForRun: [String]
```
Comment (rochaeb) [inline packages/coinstac-api-server/src/data/shared-fields.js]: should be array of ID here

File: packages/coinstac-ui/app/render/components/maps/maps-edit.jsx [hunk 2]
```diff
@@ -266,11 +268,20 @@ class MapsEdit extends Component {
...
+
+    if (mappedForRun.indexOf(currentUserId) === -1) {
+      mappedForRun = [...mappedForRun, currentUserId]
+    }
 
     this.props.getCollectionFiles(cons.id)
       .then((filesArray) => {
         this.setState({isMapped: true});
 
+        this.props.saveConsortium({ id: cons.id, name: cons.name, description: cons.description, mappedForRun });
```
Comment (rochaeb) [inline packages/coinstac-ui/app/render/components/maps/maps-edit.jsx]: no need to send name and description here

Comment (kalitopix) [reply packages/coinstac-ui/app/render/components/maps/maps-edit.jsx]: @rochaeb , name and description are required fields for consortia

Comment (rochaeb) [reply packages/coinstac-ui/app/render/components/maps/maps-edit.jsx]: So I'd say create another mutation for this, to improve readability. Using this mutation as is can be misleading as you're sending unnecessary stuff...

**Comment by rochaeb** [comment]
it's looking good, but I think that unmapping only when another pipeline is set is not enough... you'd definitely need to unmap if a user gets kicked out of a consortia or chooses to leave the consortia. And also maybe on user logout? @rssk, what do you think?

**Comment by rssk** [comment]
I think we talked about this a bit over slack, but I think all the unmapping should be moved to one call, otherwise this is going to get out of hand. I think for now, I'd like that to do both local and remote map/unmap, it really isnt any more overhead and can always be changed later based on our local store reqs. What do you think @rochaeb @kalitopix

**Review by rssk — CHANGES_REQUESTED** [review]
Add a sane way to unmap local an remote

File: packages/coinstac-ui/app/render/components/maps/maps-edit.jsx (hunk 2 above)
Comment (kalitopix) [reply packages/coinstac-ui/app/render/components/maps/maps-edit.jsx]: @rochaeb , do you think creating another mutation is better than sending unnecessary info?
what about you, @rssk ?

**Comment by kalitopix** [comment]
@rssk , @rochaeb 
I created a mutation that updates the mapping status of consortium.

Do we need to unmap user if he gets kicked out of a consortia or chooses to leave the consortia or maybe on logout as @rochaeb suggested?

File: packages/coinstac-ui/app/render/components/maps/maps-edit.jsx (hunk 2 above)
Comment (rochaeb) [reply packages/coinstac-ui/app/render/components/maps/maps-edit.jsx]: yeah, I think so

Comment (kalitopix) [reply packages/coinstac-ui/app/render/components/maps/maps-edit.jsx]: @rochaeb , it is already done

**Review by rssk — APPROVED** [review]
revisions look good!

**Pull request merged**